slime/backends/utils/data.py

```diff
@@ -1,47 +1,21 @@
-from typing import Any, Optional
+from typing import Optional
 
 import ray
 import torch
 import torch.distributed as dist
 
-from slime.utils.memory_utils import clear_memory
 from slime.utils.seqlen_balancing import get_seqlen_balanced_partitions
 from slime.utils.timer import Timer
 
 
-LOCAL_STORAGE = {}
-LOCAL_METADATA = {}
-
-
-def set_local_storage(key: str, value: Any):
-    LOCAL_STORAGE[key] = value
-
-
-def get_local_storage(key: Optional[str] = None):
-    if key is None:
-        return LOCAL_STORAGE
-    return LOCAL_STORAGE.get(key, None)
-
-
-def clear_local_storage():
-    LOCAL_STORAGE.clear()
-    clear_memory()
-
-
-def set_metadata(key: str, value: Any):
-    LOCAL_METADATA[key] = value
-
-
-def get_metadata(key: Optional[str] = None):
-    return LOCAL_METADATA.get(key, None)
-
-
 class DataIterator:
     def __init__(
         self,
+        local_storage,
```

Comment:
nit: wondering whether renaming `local_storage` will be a tiny little bit better